server/src/user/user.service.ts

```diff
@@ -113,37 +113,49 @@ export class UserService {
     }
   }
 
-  async deleteUser(userId: number, houseId: number) {
+  async deleteUser(userId: number) {
     try {
       const userHouse = await this.prisma.userHouse.findFirst({
-        where: { userId, houseId },
+        where: { userId },
       });
 
       if (!userHouse) {
-        throw new NotFoundException(
-          `User house not found for user ID ${userId} and house ID ${houseId}`,
-        );
+        await this.deleteUserFromDatabase(userId);
+        return { message: 'User deleted successfully.' };
       }
 
       if (userHouse.isAdmin) {
-        const otherAdminsCount = await this.prisma.userHouse.count({
-          where: { houseId, isAdmin: true, userId: { not: userId } },
-        });
-        if (otherAdminsCount === 0) {
-          throw new ForbiddenException(
-            'Cannot delete account. You are the only admin in this house.',
-          );
-        }
+        await this.checkAdminDeletionPossibility(userHouse);
       }
 
-      await this.prisma.user.delete({
-        where: { id: userId },
-      });
+      await this.deleteUserFromDatabase(userId);
 
       return { message: 'User deleted successfully.' };
     } catch (error) {
       console.error('Error deleting user:', error);
       throw error;
     }
   }
+
+  private async deleteUserFromDatabase(userId: number) {
+    await this.prisma.user.delete({
+      where: { id: userId },
+    });
+  }
+
+  private async checkAdminDeletionPossibility(userHouse: any) {
```

Comment:
Could you avoid using `any` here? I also prefer to make a function in case we use it in some places (not only one) or if it is too complicated. So, I think you don't have to make functionality here, especially considering the consistency between the controller and service file. We usually only make the function for a controller file except for the above reason (using many places or the logic is too complicated).

Reply:
I see! I moved the function inside of `deleteUser`.
ae02761801a3867e730035424e46cd2ade8f2087

Reply:
Oh do you want me to move the `deleteUserFromDatabase` function, too? It's needed 2times, so I made a separate function. You think it's better to put this inside of `deleteUser` using 2 times?